lib/KTable/index.vue

```diff
@@ -294,156 +294,144 @@
-                  } else if (rowIndex < totalRows - 1) {
-                    nextColIndex = 0;
-                    nextRowIndex = rowIndex + 1;
-                  } else {
-                    // Allow default behavior when reaching the last cell
-                    return;
-                  }
-                }
-              } else {
-                if (focusedElementIndex < focusableElements.length - 1) {
-                  // if navigating between more focusable elements within the cell
-                  focusableElements[focusedElementIndex + 1].focus();
-                  event.preventDefault();
-                  return;
-                } else {
-                  if (colIndex > 0) {
-                    nextColIndex = colIndex - 1;
-                  } else if (rowIndex > 0) {
-                    nextColIndex = totalCols - 1;
-                    nextRowIndex = rowIndex - 1;
-                  } else {
-                    // Allow default behavior when reaching the first cell
-                    return;
-                  }
-                }
-              }
+        if (!event.shiftKey) {
+          // Tab key navigation
+          if (focusedIndex < cellAndFocusableElements.length - 1) {
+            // Move to the next focusable element within the cell
+            cellAndFocusableElements[focusedIndex + 1].focus();
+            event.preventDefault();
+            return;
+          } else {
+            // Move to the first focusable element of the next cell
+            if (colIndex < totalCols - 1) {
+              nextColIndex = colIndex + 1;
+            } else if (rowIndex < totalRows - 1) {
+              nextColIndex = 0;
+              nextRowIndex = rowIndex + 1;
             } else {
-              if (!event.shiftKey) {
-                if (colIndex < totalCols - 1) {
-                  nextColIndex = colIndex + 1;
-                } else if (rowIndex < totalRows - 1) {
-                  nextColIndex = 0;
-                  nextRowIndex = rowIndex + 1;
-                } else {
-                  // Allow default behavior when reaching the last cell
-                  return;
-                }
-              } else {
-                if (colIndex > 0) {
-                  nextColIndex = colIndex - 1;
-                } else if (rowIndex > 0) {
-                  nextColIndex = totalCols - 1;
-                  nextRowIndex = rowIndex - 1;
-                } else {
-                  // Allow default behavior when reaching the first cell
-                  return;
-                }
-              }
+              return; // Allow default Tab behavior when reaching the end
             }
-
-            break;
+            const nextCell = this.getCell(nextRowIndex, nextColIndex);
+            const nextFocusableElements = this.getFocusableElements(nextCell);
+            const nextCellAndFocusableElements = [nextCell, ...nextFocusableElements];
+            nextCellAndFocusableElements[0].focus();
```

Comment:
Why do we need this `nextCellAndFocusableElements` if we always will be just focusing the index 0? We dont do anything with this array after that. That means that we will be always focusing the nextCell. We dont need to query the focusable elments of the next cell.

Reply:
Btw in the `updateFocusState` we are already focusing the next cell, so there is no need of any of these lines:

```javascript
            const nextCell = this.getCell(nextRowIndex, nextColIndex);
            const nextFocusableElements = this.getFocusableElements(nextCell);
            const nextCellAndFocusableElements = [nextCell, ...nextFocusableElements];
            nextCellAndFocusableElements[0].focus();
```